File: resource/doc/v1/device/product.js
/**
 * @api {post} /v1/device/product/getProductList 商品列表
 * @apiVersion 0.0.1
 * @apiName 商品列表
 * @apiGroup 商品管理
 * @apiPermission device
 *
 * @apiDescription 获取商品列表数据
 *     
 * @apiParam {Number} page 页码
 * @apiParam {Number} pageSize 每页数量
 * @apiParam {Number} cateId 商户分类 id
 * @apiParam {Number} tenancyCategoryId 商城分类 id 
 * @apiParam {String} keyword 关键字搜索 
 * @apiParam {String} type 商品类型 1.普通商品 2.秒杀商品,3.预售商品，4.助力商品
 * @apiParam {String} [isGiftBag] 是否礼包 1 是 ，2 否
 * 
 * @apiBody {Number} page 页码
 * @apiBody {Number} pageSize 每页数量
 * @apiBody {Number} cateId 商户分类 id
 * @apiBody {Number} tenancyCategoryId 商城分类 id 
 * @apiBody {String} keyword 关键字搜索 
 * @apiBody {String} type 商品类型 1.普通商品 2.秒杀商品,3.预售商品，4.助力商品
 * @apiBody {String} [isGiftBag] 是否礼包 1 是 ，2 否
 *
 * @apiHeader {String} Authorization 接口需要带上此头信息
 * @apiHeaderExample {Header} Header-Example
 *     "Authorization: Bearer 5f048fe"
 *
 * @apiExample {bash} Curl example
 * curl -H "Authorization: Bearer 5f048fe" -i http://10.0.0.26:8085/v1/device/product/getProductList
 *
 * @apiUse TokenError
 *
 * @apiSuccessExample Response:
 *     HTTP/1.1 200 OK
 *     {
    "status": 200,
    "data": {
        "list": [
            {
                "id": 1,
                "storeName": "领立裁腰带短袖连衣裙",
                "sales": 1,
                "price": 80,
                "image": "http://127.0.0.1:8090/uploads/def/20200816/9a6a2e1231fb19517ed1de71206a0657.jpg"
            }
        ],
        "total": 1,
        "page": 1,
        "pageSize": 10
    },
    "message": "获取成功"
 *     }
 */




/**
 * @api {get} /v1/device/product/getProductById/1 商品详情
 * @apiVersion 0.0.1
 * @apiName 商品详情
 * @apiGroup 商品管理
 * @apiPermission device
 *
 * @apiDescription 获取商品详情数据
 *     
 *
 * @apiHeader {String} Authorization 接口需要带上此头信息
 * @apiHeaderExample {Header} Header-Example
 *     "Authorization: Bearer 5f048fe"
 *
 * @apiExample {bash} Curl example
 * curl -H "Authorization: Bearer 5f048fe" -i http://10.0.0.26:8085/v1/device/product/getProductById/1
 *
 * @apiSuccess {Number}   id            商品id
 * @apiSuccess {String}   storeName     商品名称
 * @apiSuccess {String}   storeInfo     商品简介
 * @apiSuccess {String}   keyword       商品关键词
 * @apiSuccess {String}   barCode       商品条码
 * @apiSuccess {Number}   isShow        是否显示 1 是，2否
 * @apiSuccess {Number}   status        商品状态 1：审核通过,2：审核中 3: 未通过
 * @apiSuccess {String}   unitName      商品单位
 * @apiSuccess {Number}   sort      排序
 * @apiSuccess {Number}   rank      总后台排序
 * @apiSuccess {Number}   sales      商品销量
 * @apiSuccess {Number}   price      最低价格
 * @apiSuccess {Number}   cost      成本价
 * @apiSuccess {Number}   otPrice      原价
 * @apiSuccess {Number}   stock      库存
 * @apiSuccess {Number}   isHot      是否热卖 1 是，2否
 * @apiSuccess {Number}   isBenefit      促销单品 1 是，2否
 * @apiSuccess {Number}   isBest      是否精品 1 是，2否
 * @apiSuccess {Number}   isNew      是否新品 1 是，2否
 * @apiSuccess {Number}   isGood      是否优品推荐 1 是，2否
 * @apiSuccess {Number}   productType      商品分类 1.普通商品 2.秒杀商品,3.预售商品，4.助力商品
 * @apiSuccess {Number}   ficti      虚拟销量
 * @apiSuccess {Number}   browse      浏览量
 * @apiSuccess {String}   codePath      产品二维码地址
 * @apiSuccess {String}   videoLink      主图视频链接
 * @apiSuccess {Number}   specType       规格 1单 2多
 * @apiSuccess {String}   refusal       审核拒绝理由
 * @apiSuccess {Number}   Rate       评价分数
 * @apiSuccess {Number}   ReplyCount       评论数
 * @apiSuccess {Number}   isGiftBag       是否为礼包
 * @apiSuccess {Number}   careCount       收藏数
 * @apiSuccess {String}   image       商品图片
 * @apiSuccess {String}   sysTenancyName       医院名称
 * @apiSuccess {String}   cateName       后台分类名称
 * @apiSuccess {String}   brandName       品牌名称
 * @apiSuccess {String}   tempName       模板名称
 * @apiSuccess {String}   content       详情内容
 * @apiSuccess {String[]}   sliderImages      展示图片
 * @apiSuccess {String[]}   attr      规格
 * @apiSuccess {Object[]}   attrValue      规格详情
 * @apiSuccess {Object[]}   productCates  商品分类
 *
 * @apiUse TokenError
 * 
 * @apiSuccessExample Response:
 *     HTTP/1.1 200 OK
 *     {
  "status": 200,
    "data": {
        "id": 1,
        "storeName": "领立裁腰带短袖连衣裙",
        "storeInfo": "短袖连衣裙",
        "keyword": "连衣裙",
        "unitName": "件",
        "sort": 40,
        "sales": 1,
        "price": 80,
        "otPrice": 100,
        "stock": 399,
        "isHot": 2,
        "isBenefit": 2,
        "isBest": 2,
        "isNew": 2,
        "isGood": 1,
        "productType": 2,
        "ficti": 100,
        "specType": 1,
        "rate": 5,
        "isGiftBag": 2,
        "image": "http://127.0.0.1:8090/uploads/def/20200816/9a6a2e1231fb19517ed1de71206a0657.jpg",
        "tempId": 99,
        "sysTenancyId": 1,
        "sysBrandId": 2,
        "productCategoryId": 162,
        "sysTenancyName": "宝安中心人民医院",
        "cateName": "男士上衣",
        "brandName": "苹果",
        "tempName": "",
        "content": "<p>好手机</p>",
        "sliderImage": "http://127.0.0.1:8090/uploads/def/20200816/9a6a2e1231fb19517ed1de71206a0657.jpg,http://127.0.0.1:8090/uploads/def/20200816/9a6a2e1231fb19517ed1de71206a0657.jpg",
        "sliderImages": [
            "http://127.0.0.1:8090/uploads/def/20200816/9a6a2e1231fb19517ed1de71206a0657.jpg",
            "http://127.0.0.1:8090/uploads/def/20200816/9a6a2e1231fb19517ed1de71206a0657.jpg"
        ],
        "attr": [
            {
                "detail": [
                    "35"
                ],
                "value": "S"
            },
            {
                "detail": [
                    "36"
                ],
                "value": "L"
            },
            {
                "detail": [
                    "37"
                ],
                "value": "XL"
            },
            {
                "detail": [
                    "38"
                ],
                "value": "XXL"
            }
        ],
        "attrValue": [
            {
                "sku": "S",
                "stock": 99,
                "sales": 1,
                "image": "\thttp://127.0.0.1:8090/uploads/def/20200816/9a6a2e1231fb19517ed1de71206a0657.jpg",
                "barCode": "123456",
                "cost": 50,
                "otPrice": 180,
                "price": 160,
                "volume": 1,
                "weight": 1,
                "extensionOne": 0,
                "extensionTwo": 0,
                "unique": "e2fe28308fd0",
                "detail": {
                    "尺寸": "S"
                },
                "value0": "S"
            },
            {
                "sku": "L",
                "stock": 100,
                "sales": 0,
                "image": "\thttp://127.0.0.1:8090/uploads/def/20200816/9a6a2e1231fb19517ed1de71206a0657.jpg",
                "barCode": "123456",
                "cost": 50,
                "otPrice": 180,
                "price": 160,
                "volume": 1,
                "weight": 1,
                "extensionOne": 0,
                "extensionTwo": 0,
                "unique": "e2fe28308fd0",
                "detail": {
                    "尺寸": "L"
                },
                "value0": "L"
            },
            {
                "sku": "XL",
                "stock": 100,
                "sales": 0,
                "image": "\thttp://127.0.0.1:8090/uploads/def/20200816/9a6a2e1231fb19517ed1de71206a0657.jpg",
                "barCode": "123456",
                "cost": 50,
                "otPrice": 180,
                "price": 160,
                "volume": 1,
                "weight": 1,
                "extensionOne": 0,
                "extensionTwo": 0,
                "unique": "e2fe28308fd0",
                "detail": {
                    "尺寸": "XL"
                },
                "value0": "XL"
            },
            {
                "sku": "XXL",
                "stock": 100,
                "sales": 0,
                "image": "\thttp://127.0.0.1:8090/uploads/def/20200816/9a6a2e1231fb19517ed1de71206a0657.jpg",
                "barCode": "123456",
                "cost": 50,
                "otPrice": 180,
                "price": 160,
                "volume": 1,
                "weight": 1,
                "extensionOne": 0,
                "extensionTwo": 0,
                "unique": "e2fe28308fd0",
                "detail": {
                    "尺寸": "XXL"
                },
                "value0": "XXL"
            }
        ],
        "cateId": 162,
        "tenancyCategoryId": [
            174,
            173
        ],
        "productCates": [
            {
                "id": 174,
                "cateName": "时尚女装"
            },
            {
                "id": 173,
                "cateName": "品牌服饰"
            }
        ]
    },
    "message": "操作成功"
 *     }
 */

File: source/product.go
package source

import (
	"time"

	"github.com/gookit/color"
	"github.com/snowlyg/go-tenancy/g"
	"github.com/snowlyg/go-tenancy/model"
	"gorm.io/gorm"
)

var Product = new(product)

type product struct{}

// 出售中 1: is_show' => 1, 'status' => 1
// 仓库中 2:'is_show' => 2, 'status' => 1
// 3,4,5 商户才有
// 已售罄 3:'is_show' => 1, 'stock' => 0, 'status' => 1
// 警戒库存 4:'stock' => $stock ? $stock : 0, 'status' => 1
// 回收站 5:'deleted_at' => not null
// 待审核 6:'status' => 2
// 审核未通过 7:'status' => 3
var products = []model.Product{
	{BaseProduct: model.BaseProduct{SysTenancyID: 1, StoreName: "领立裁腰带短袖连衣裙", StoreInfo: "短袖连衣裙", Keyword: "连衣裙", BarCode: "", SysBrandID: 2, IsShow: g.StatusTrue, Status: model.SuccessProductStatus, ProductCategoryID: 176, UnitName: "件", Sort: 40, Rank: 0, Sales: 1, Price: 80.00, Cost: 50.00, OtPrice: 100.00, Stock: 399, IsHot: g.StatusFalse, IsBenefit: g.StatusFalse, IsBest: g.StatusFalse, IsNew: g.StatusFalse, IsGood: g.StatusTrue, ProductType: model.GeneralSale, Ficti: 100, Browse: 0, CodePath: "", VideoLink: "", TempID: 99, SpecType: model.DoubleSpec, Refusal: "", Rate: 5.0, ReplyCount: 0, IsGiftBag: g.StatusFalse, CareCount: 0, OldID: 0, Image: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"}, SliderImage: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg,http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"},

	{BaseProduct: model.BaseProduct{SysTenancyID: 1, StoreName: "纯棉珠地撞色领polo裙", StoreInfo: "polo裙", Keyword: "polo裙", BarCode: "", SysBrandID: 2, IsShow: g.StatusTrue, Status: model.SuccessProductStatus, ProductCategoryID: 176, UnitName: "件", Sort: 40, Rank: 0, Sales: 1, Price: 160.00, Cost: 50.00, OtPrice: 180.00, Stock: 99, IsHot: g.StatusFalse, IsBenefit: g.StatusFalse, IsBest: g.StatusFalse, IsNew: g.StatusFalse, IsGood: g.StatusTrue, ProductType: model.GeneralSale, Ficti: 100, Browse: 0, CodePath: "", VideoLink: "", TempID: 99, SpecType: model.SingleSpec, Refusal: "", Rate: 5.0, ReplyCount: 0, IsGiftBag: g.StatusFalse, CareCount: 0, OldID: 0, Image: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"}, SliderImage: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg,http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"},

	{BaseProduct: model.BaseProduct{SysTenancyID: 1, StoreName: "精梳棉修身短袖T恤（圆/V领）", StoreInfo: "精梳", Keyword: "T恤", BarCode: "", SysBrandID: 2, IsShow: g.StatusTrue, Status: model.SuccessProductStatus, ProductCategoryID: 176, UnitName: "件", Sort: 40, Rank: 0, Sales: 2, Price: 40.00, Cost: 20.00, OtPrice: 58.00, Stock: 0, IsHot: g.StatusFalse, IsBenefit: g.StatusFalse, IsBest: g.StatusFalse, IsNew: g.StatusFalse, IsGood: g.StatusTrue, ProductType: model.GeneralSale, Ficti: 100, Browse: 0, CodePath: "", VideoLink: "", TempID: 102, SpecType: model.SingleSpec, Refusal: "", Rate: 5.0, ReplyCount: 0, IsGiftBag: g.StatusFalse, CareCount: 0, OldID: 0, Image: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"}, SliderImage: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg,http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"},

	{BaseProduct: model.BaseProduct{SysTenancyID: 1, StoreName: "素湃黑科技纯棉疏水抗污短袖T恤", StoreInfo: "黑科技", Keyword: "T恤", BarCode: "", SysBrandID: 2, IsShow: g.StatusTrue, Status: model.AuditProductStatus, ProductCategoryID: 176, UnitName: "件", Sort: 0, Rank: 0, Sales: 1, Price: 80.00, Cost: 60.00, OtPrice: 100.00, Stock: 99, IsHot: g.StatusFalse, IsBenefit: g.StatusFalse, IsBest: g.StatusFalse, IsNew: g.StatusFalse, IsGood: g.StatusTrue, ProductType: model.GeneralSale, Ficti: 100, Browse: 0, CodePath: "", VideoLink: "", TempID: 99, SpecType: model.SingleSpec, Refusal: "", Rate: 5.0, ReplyCount: 0, IsGiftBag: g.StatusFalse, CareCount: 0, OldID: 0, Image: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"}, SliderImage: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg,http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"},

	{BaseProduct: model.BaseProduct{SysTenancyID: 1, StoreName: "智能定制休闲单西 布雷泽海军蓝轻薄斜纹", StoreInfo: "西装定制", Keyword: "西装", BarCode: "", SysBrandID: 2, IsShow: g.StatusTrue, Status: model.FailProductStatus, ProductCategoryID: 176, UnitName: "件", Sort: 70, Rank: 0, Sales: 3, Price: 880.00, Cost: 500.00, OtPrice: 1680.00, Stock: 97, IsHot: g.StatusFalse, IsBenefit: g.StatusFalse, IsBest: g.StatusFalse, IsNew: g.StatusFalse, IsGood: g.StatusTrue, ProductType: model.GeneralSale, Ficti: 100, Browse: 0, CodePath: "", VideoLink: "", TempID: 99, SpecType: model.SingleSpec, Refusal: "", Rate: 5.0, ReplyCount: 0, IsGiftBag: g.StatusFalse, CareCount: 0, OldID: 0, Image: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"}, SliderImage: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg,http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"},

	{TENANCY_MODEL: g.TENANCY_MODEL{DeletedAt: gorm.DeletedAt{Time: time.Now(), Valid: true}}, BaseProduct: model.BaseProduct{SysTenancyID: 1, StoreName: "梅湾街复古雪纺翻领上衣", StoreInfo: "雪纺", Keyword: "上衣", BarCode: "", SysBrandID: 2, IsShow: g.StatusTrue, Status: model.SuccessProductStatus, ProductCategoryID: 176, UnitName: "件", Sort: 56, Rank: 0, Sales: 1, Price: 88.00, Cost: 100.00, OtPrice: 200.00, Stock: 134, IsHot: g.StatusFalse, IsBenefit: g.StatusFalse, IsBest: g.StatusFalse, IsNew: g.StatusFalse, IsGood: g.StatusTrue, ProductType: model.GeneralSale, Ficti: 100, Browse: 0, CodePath: "", VideoLink: "", TempID: 96, SpecType: model.SingleSpec, Refusal: "", Rate: 5.0, ReplyCount: 0, IsGiftBag: g.StatusFalse, CareCount: 4, OldID: 0, Image: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"}, SliderImage: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg,http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"},

	{BaseProduct: model.BaseProduct{SysTenancyID: 1, StoreName: "天丝一字领露肩上衣", StoreInfo: "天丝一字领露肩上衣", Keyword: "测试", BarCode: "", SysBrandID: 2, IsShow: g.StatusTrue, Status: model.SuccessProductStatus, ProductCategoryID: 176, UnitName: "件", Sort: 10, Rank: 0, Sales: 1, Price: 50.00, Cost: 20.00, OtPrice: 100.00, Stock: 99, IsHot: g.StatusFalse, IsBest: g.StatusFalse, IsNew: g.StatusFalse, IsGood: g.StatusTrue, ProductType: model.GeneralSale, Ficti: 100, Browse: 0, CodePath: "", VideoLink: "", TempID: 102, SpecType: model.SingleSpec, Refusal: "", Rate: 5.0, ReplyCount: 0, IsGiftBag: g.StatusFalse, CareCount: 1, OldID: 0, Image: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"}, TENANCY_MODEL: g.TENANCY_MODEL{DeletedAt: gorm.DeletedAt{Time: time.Now()}}, SliderImage: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg,http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg"},
}

var productContents = []model.ProductContent{
	{Content: "<p>好手机</p>", ProductID: 1, Type: 1},
	{Content: "<p>好手机</p>", ProductID: 2, Type: 1},
	{Content: "<p>好手机</p>", ProductID: 3, Type: 1},
	{Content: "<p>好手机</p>", ProductID: 4, Type: 1},
	{Content: "<p>好手机</p>", ProductID: 5, Type: 1},
	{Content: "<p>好手机</p>", ProductID: 6, Type: 1},
	{Content: "<p>好手机</p>", ProductID: 7, Type: 1},
}
var productAttrValues = []model.ProductAttrValue{
	{ProductID: 2, Detail: "", BaseProductAttrValue: model.BaseProductAttrValue{Sku: "", Stock: 99, Sales: 1, Image: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg", BarCode: "12444", Cost: 50.00, OtPrice: 180.00, Price: 160.00, Volume: 1.00, Weight: 1.00, ExtensionOne: 0.00, ExtensionTwo: 0.00, Unique: "e2fe28308fd0"}, Type: 1},
	{ProductID: 3, Detail: "", BaseProductAttrValue: model.BaseProductAttrValue{Sku: "", Stock: 99, Sales: 1, Image: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg", BarCode: "12444", Cost: 50.00, OtPrice: 180.00, Price: 160.00, Volume: 1.00, Weight: 1.00, ExtensionOne: 0.00, ExtensionTwo: 0.00, Unique: "e2fe28308fd0"}, Type: 1},
	{ProductID: 4, Detail: "", BaseProductAttrValue: model.BaseProductAttrValue{Sku: "", Stock: 99, Sales: 1, Image: "	http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg", BarCode: "15454545", Cost: 50.00, OtPrice: 180.00, Price: 160.00, Volume: 1.00, Weight: 1.00, ExtensionOne: 0.00, ExtensionTwo: 0.00, Unique: "e2fe28308fd1"}, Type: 1},
	{ProductID: 5, Detail: "", BaseProductAttrValue: model.BaseProductAttrValue{Sku: "", Stock: 99, Sales: 1, Image: "	http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg", BarCode: "15454545", Cost: 50.00, OtPrice: 180.00, Price: 160.00, Volume: 1.00, Weight: 1.00, ExtensionOne: 0.00, ExtensionTwo: 0.00, Unique: "e2fe28308fd1"}, Type: 1},
	{ProductID: 1, Detail: "{\"\u5c3a\u5bf8\": \"S\"}", BaseProductAttrValue: model.BaseProductAttrValue{Sku: "S", Stock: 99, Sales: 1, Image: "	http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg", BarCode: "123456", Cost: 50.00, OtPrice: 180.00, Price: 160.00, Volume: 1.00, Weight: 1.00, ExtensionOne: 0.00, ExtensionTwo: 0.00, Unique: "e2fe28308fd2"}, Type: 1},
	{ProductID: 1, Detail: "{\"\u5c3a\u5bf8\": \"L\"}", BaseProductAttrValue: model.BaseProductAttrValue{Sku: "L", Stock: 100, Sales: 0, Image: "	http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg", BarCode: "123456", Cost: 50.00, OtPrice: 180.00, Price: 160.00, Volume: 1.00, Weight: 1.00, ExtensionOne: 0.00, ExtensionTwo: 0.00, Unique: "e2fe28308fd3"}, Type: 1},
	{ProductID: 1, Detail: "{\"\u5c3a\u5bf8\": \"XL\"}", BaseProductAttrValue: model.BaseProductAttrValue{Sku: "XL", Stock: 100, Sales: 0, Image: "	http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg", BarCode: "123456", Cost: 50.00, OtPrice: 180.00, Price: 160.00, Volume: 1.00, Weight: 1.00, ExtensionOne: 0.00, ExtensionTwo: 0.00, Unique: "e2fe28308fd4"}, Type: 1},
	{ProductID: 1, Detail: "{\"\u5c3a\u5bf8\": \"XXL\"}", BaseProductAttrValue: model.BaseProductAttrValue{Sku: "XXL", Stock: 100, Sales: 0, Image: "	http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg", BarCode: "123456", Cost: 50.00, OtPrice: 180.00, Price: 160.00, Volume: 1.00, Weight: 1.00, ExtensionOne: 0.00, ExtensionTwo: 0.00, Unique: "e2fe28308fd5"}, Type: 1},
	{ProductID: 6, Detail: "", BaseProductAttrValue: model.BaseProductAttrValue{Sku: "", Stock: 99, Sales: 1, Image: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg", BarCode: "1774575", Cost: 50.00, OtPrice: 180.00, Price: 160.00, Volume: 1.00, Weight: 1.00, ExtensionOne: 0.00, ExtensionTwo: 0.00, Unique: "e2fe28308fd6"}, Type: 1},
	{ProductID: 7, Detail: "", BaseProductAttrValue: model.BaseProductAttrValue{Sku: "", Stock: 98, Sales: 2, Image: "http://127.0.0.1:8089/uploads/file/9a6a2e1231fb19517ed1de71206a0657.jpg", BarCode: "10024242", Cost: 50.00, OtPrice: 180.00, Price: 160.00, Volume: 1.00, Weight: 1.00, ExtensionOne: 0.00, ExtensionTwo: 0.00, Unique: "e2fe28308fd7"}, Type: 1},
}

func (m *product) Init() error {
	return g.TENANCY_DB.Transaction(func(tx *gorm.DB) error {
		if tx.Where("id IN ?", []int{1}).Find(&[]model.Product{}).RowsAffected == 1 {
			color.Danger.Println("\n[Mysql] --> sys_products 表的初始数据已存在!")
			return nil
		}
		if err := tx.Create(&products).Error; err != nil { // 遇到错误时回滚事务
			return err
		}
		if err := tx.Model(&model.ProductContent{}).Create(&productContents).Error; err != nil { // 遇到错误时回滚事务
			return err
		}
		if err := tx.Model(&model.ProductAttrValue{}).Create(&productAttrValues).Error; err != nil { // 遇到错误时回滚事务
			return err
		}
		color.Info.Println("\n[Mysql] --> sys_products 表初始数据成功!")
		return nil
	})
}
